controllers/auth_config_status_updater.go

```diff
@@ -41,41 +46,152 @@ func (u *AuthConfigStatusUpdater) Reconcile(ctx context.Context, req ctrl.Reques
+
+	var condition api.Condition
+	linked, loose := u.partitionHostsByStatus(cacheId, authConfig)
+	if len(loose) > 0 {
+		condition = statusNotReady(api.StatusReasonHostNotLinked, "One or more hosts not linked to the resource")
+	} else {
+		condition = statusReady()
 	}
 
-	authConfig.Status.Ready = ready
-	authConfig.Status.NumIdentitySources = int64(len(authConfig.Spec.Identity))
-	authConfig.Status.NumMetadataSources = int64(len(authConfig.Spec.Metadata))
-	authConfig.Status.NumAuthorizationPolicies = int64(len(authConfig.Spec.Authorization))
-	authConfig.Status.NumResponseItems = int64(len(authConfig.Spec.Response))
+	var conditionsChanged, summaryChanged bool
 
-	issuingWristbands := false
-	for _, responseConfig := range authConfig.Spec.Response {
-		if responseConfig.GetType() == api.ResponseWristband {
-			issuingWristbands = true
-			break
-		}
+	authConfig.Status.Conditions, conditionsChanged = buildStatusConditions(authConfig.Status.Conditions, condition)
+	authConfig.Status.Summary, summaryChanged = buildStatusSummary(authConfig, linked)
+
+	if !authConfig.Status.Ready() {
+		err = fmt.Errorf("resource not ready")
 	}
-	authConfig.Status.FestivalWristbandEnabled = issuingWristbands
 
-	return u.Status().Update(ctx, authConfig)
+	if changed := conditionsChanged || summaryChanged; !changed {
+		logger.V(1).Info("resource status did not change")
+		return // to save an update request
+	}
+
+	logger.V(1).Info("resource status changed", "authconfig/status", authConfig.Status)
+
+	if updateErr := u.Status().Update(ctx, authConfig); updateErr != nil {
+		logger.Error(updateErr, "failed to update the resource")
+		err = updateErr
+	}
+
+	logger.Info("resource status updated")
+
+	return
+}
+
+func (u *AuthConfigStatusUpdater) partitionHostsByStatus(cacheId string, authConfig *api.AuthConfig) ([]string, []string) {
+	type obj struct{}
+	hosts := make(map[string]obj)
+	for _, host := range u.Cache.FindKeys(cacheId) {
+		hosts[host] = obj{}
+	}
+	linked := []string{}
+	loose := []string{}
+	for _, host := range authConfig.Spec.Hosts {
+		if _, ok := hosts[host]; ok {
+			linked = append(linked, host)
+		} else {
+			loose = append(loose, host)
+		}
+	}
+	return linked, loose
 }
 
 func (u *AuthConfigStatusUpdater) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&api.AuthConfig{}, builder.WithPredicates(LabelSelectorPredicate(u.LabelSelector))).
 		Complete(u)
 }
+
+func statusReady() api.Condition {
+	return api.Condition{
+		Type:   api.StatusConditionReady,
+		Status: k8score.ConditionTrue,
+		Reason: api.StatusReasonReconciled,
+	}
+}
+
+func statusNotReady(reason, message string) api.Condition {
+	return api.Condition{
+		Type:    api.StatusConditionReady,
+		Status:  k8score.ConditionFalse,
+		Reason:  reason,
+		Message: message,
+	}
+}
+
+func buildStatusConditions(current []api.Condition, new api.Condition) ([]api.Condition, bool) {
+	new.LastTransitionTime = metav1.Now()
```

Comment:
nit: might be worth naming this something like ```next``` rather than new that is also a keyword in go?

Reply:
IDK if it's really "next". The semantics is: "given an existing (current) list of conditions, ensure this one is in the list".

It's more like an "add or update", copied from [here](https://github.com/Kuadrant/authorino-operator/blob/4bc68ef986c4d0ab463ff2ccabcb71d7d54fe0d4/pkg/condition/condition.go#L11), where the function is called – guess what? – "AddOrUpdateStatusConditions". Should I have stuck with that instead of "buildStatusConditions"?

Reply:
Renamed to `updateStatusConditions(currentConditions []api.Condition, newCondition api.Condition)`.